testgen/generators/collation_short.py

```diff
@@ -78,8 +92,128 @@ def insert_collation_header(self, test_objs):
+        locale = ""
+        strength = None
+
+    def parse_compare(self, line_index, lines):
+        # Handles lines in a compare region
+        # Test sections ("* compare") are terminated by
+        # definitions of new collators, changing attributes, or new test sections.
+        tests = []
+        line_in = lines[line_index]
+        if not self.compare_pattern.match(line_in):
+            return None, line_index
+
+        string1 = ''
+        line_index += 1
+        while line_index < len(lines):
+            line_in = lines[line_index]
+
+            if self.compare_pattern.match(line_in):
+                break
+            if self.locale_string.match(line_in):
+                break
+            if self.root_locale.match(line_in):
+                break
+            if self.attribute_test.match(line_in):
+                break
+            if self.reorder_test.match(line_in):
+                break
+            if self.rule_header_pattern.match(line_in):
+                break
+            if self.test_line.match(line_in):
+                break
+
+            # It's a blank line, comment, or a comparison
+            # ignore comment at end of this rulrule_match.group(1)e line
+            if line_in == '' or line_in[0] == '#':
+                # Skip blank lines
+                line_index += 1
+                continue
+
+            is_comparison_match = self.comparison_line.match(line_in)
+            if is_comparison_match:
+                compare_type = is_comparison_match.group(1)
+                # TODO !!! Check string2 for \u vs \U inm the line
+                raw_string2 = is_comparison_match.group(3)
+                string2 = ''
+                try:
+                    string2 = raw_string2.encode().decode("unicode_escape")
+                except:
+                    # Catch an error
+                    pass
+
+                compare_comment = is_comparison_match.group(5)
+                # ??? re-encode to get excaped version of s2?
+                new_test = {
+                    'compare_type': compare_type,
+                    's1': string1,
+                    's2': string2,
+                    'line': line_index,
+                }
+                if compare_comment:
+                    new_test['compare_comment'] = compare_comment
+
+                # Remember the previous string
+                string1 = string2
+                tests.append(new_test)
+            line_index += 1
+
+        return tests, line_index
+
+    def parse_rule(self, line_index, lines):
+        # Given the lines, process a rule and return the rule string,
+        # rule comments, and the new line index
+        # ignore comment at end of this rulrule_match.group(1)e line
+
+        rule_list = []
+        rule_comments = []
+
+        # Check if it's really the start of rules
+        line_in = lines[line_index]
+        if not self.rule_header_pattern.match(line_in):
+            return None, line_index
```

Comment:
Ditto: assert rather than conditional

Reply:
Changed the logic to use the functions to detect rules or comparisons. It not a match, return without advancing the line. If a match, then return the parsed results if t